src/OneDrive.Sdk.Authentication.Desktop/Business/AuthenticationContextWrapper.cs

```diff
@@ -54,6 +54,19 @@ public ITokenCache TokenCache
             }
         }
 
+
+        /// <summary>
+        /// Authenticates the daemon app silently"/>.
+        /// </summary>
+        /// <param name="resource">The resource to authenticate against.</param>
+        /// <param name="clientCredential">The client credential of the application.</param>
+        /// <returns>The <see cref="IAuthenticationResult"/>.</returns>
+        public async Task<IAuthenticationResult> AcquireDaemonTokenSilentAsync(string resource, ClientCredential clientCredential)
```

Comment:
Are you saying `AcquireTokenSilentAsync` throws an exception, but `AcquireTokenAsync` does not? Wouldn't it pop up a dialog? I thought the reason those methods were separate was because the first did not require user interaction.

Either way, I think adding this is fine, but perhaps the name should be different (since nothing about the token is inherently limited to daemons). Can you change the method name to `AcquireTokenAsync`?